scripts/gitlab-build.sh

```diff
@@ -296,40 +347,22 @@ case $BUILD_PLATFORM in
     strip_binaries
     calculate_checksums
     make_pkg
-    make_archive
+    prepare_artifacts
     push_binaries
     updater_push_release
     ;;
   x86_64-unknown-snap-gnu)
-    ARC="amd64"
-    EXT="snap"
-    apt install -y expect zip rhash
-    snapcraft clean
-    echo "Prepare snapcraft.yaml for build on Gitlab CI in Docker image"
-    sed -i 's/git/'"$VER"'/g' snap/snapcraft.yaml
-    if [[ "$CI_BUILD_REF_NAME" = "beta" || "$VER" == *1.9* ]];
-      then
-        sed -i -e 's/grade: devel/grade: stable/' snap/snapcraft.yaml;
-    fi
-    mv -f snap/snapcraft.yaml snapcraft.yaml
-    snapcraft -d
-    snapcraft_login=$(expect -c "
-      spawn snapcraft login
-      expect \"Email:\"
-      send \"$SNAP_EMAIL\n\"
-      expect \"Password:\"
-      send \"$SNAP_PASS\n\"
-      expect \"\$\"
-      ")
-    echo "$snapcraft_login"
-    snapcraft push "parity_"$VER"_amd64.snap"
-    snapcraft status parity
-    snapcraft logout
-    $MD5_BIN "parity_"$VER"_amd64.snap" > "parity_"$VER"_amd64.snap.md5"
-    $SHA256_BIN "parity_"$VER"_amd64.snap" > "parity_"$VER"_amd64.snap.sha256"
-    echo "add artifacts to archive"
-    rm -rf parity.zip
-    zip -r parity.zip "parity_"$VER"_amd64.snap" "parity_"$VER"_amd64.snap.md5" "parity_"$VER"_amd64.snap.sha256"
+    make_snap
+    ;;
+  i686-unknown-snap-gnu)
+    set_env
```

Comment:
We do we need `set_env` here but not for other snaps? Do we actually care about that at all?

Reply:
in this case it is necessary

Reply:
why? can you elaborate?